spirv_cross.cpp

```diff
@@ -1468,6 +1468,39 @@ bool Compiler::get_binary_offset_for_decoration(VariableID id, spv::Decoration d
 	return true;
 }
 
+bool Compiler::block_is_noop(const SPIRBlock &block) const
+{
+	if (block.terminator != SPIRBlock::Direct)
+		return false;
+
+	// Verify all instructions have no semantic impact.
+	for (auto &i : block.ops)
+	{
+		auto op = static_cast<Op>(i.op);
+
+		switch (op)
+		{
+		// Non-Semantic instructions.
+		case OpNop:
+		case OpSourceContinued:
+		case OpSource:
+		case OpSourceExtension:
+		case OpName:
+		case OpMemberName:
+		case OpString:
+		case OpLine:
+		case OpNoLine:
+		case OpModuleProcessed:
+			break;
+
+		default:
+			return false;
+		}
+	}
+
+	return true;
```

Comment:
`Compiler::execution_is_noop` checks the next block's phi variables and verifies their parents aren't the current block. Should I add that here as well?